keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/NpmArtifactSupplier.kt

```diff
@@ -71,10 +74,16 @@ class NpmArtifactSupplier(
    * Extracts the commit hash from the version string using the Netflix semver convention.
    */
   override fun getGitMetadata(artifact: PublishedArtifact, versioningStrategy: VersioningStrategy): GitMetadata? {
+
     return NetflixSemVerVersioningStrategy.getCommitHash(artifact)
       ?.let { GitMetadata(it) }
   }
 
+  override fun getArtifactMetadata(artifact: PublishedArtifact): ArtifactMetadata? {
```

Comment:
The implementation of all 3 `ArtifactSupplier` classes for this function is the same. Could we make it the default implementation in the interface? You can define the dependency on `ArtifactMetadataService` there as well.

Reply:
lets talk about this offline